File: app/src/main/java/com/biginsect/easyhub/ui/base/BaseListFragment.java
package com.biginsect.easyhub.ui.base;


import android.os.Bundle;
import android.view.View;

/**
 * @author biginsect
 * @date 2018/8/31.
 */

public abstract class BaseListFragment <V extends IBaseContract.IView, P extends IBaseContract.IPresenter<V>, A extends BaseAdapter>
        extends BaseFragment<V, P> implements IBaseContract.IView,
        BaseViewHolder.OnItemClickListener,
        BaseViewHolder.OnItemLongClickListener{
    protected A adapter;

    @Override
    protected void initFragment(Bundle savedInstanceState) {
        adapter = getAdapter();
        if (null != adapter){
            adapter.setOnItemClickListener(this);
            adapter.setOnItemClickListener(this);
        }
    }

    @Override
    public void onItemClick(int position, View view) {

    }

    @Override
    public boolean onItemLongClick(int position, View view) {
        return false;
    }

    @Override
    protected abstract int getLayoutId();

    protected abstract A getAdapter();
}

File: app/src/main/java/com/biginsect/easyhub/ui/base/BaseListFragment.kt
package com.biginsect.easyhub.ui.base

import android.os.Bundle
import android.view.View

/**
 * @author big insect
 * @date 2018/9/3.
 */

abstract class BaseListFragment<V: IBaseContract.IView, P: IBaseContract.IPresenter<V>, A: BaseAdapter<*, *>>
    :BaseFragment<V, P>(), IBaseContract.IView,
        BaseViewHolder.OnItemLongClickListener, BaseViewHolder.OnItemClickListener{

    internal var adapter: A? = null
        private set

    override fun initFragment(savedInstanceState: Bundle?) {
        adapter = getAdapter()
        if (null != adapter){
            adapter!!.onItemClickListener = this
            adapter!!.onItemLongClickListener = this
        }
    }

    override fun onItemClick(position: Int, view: View?) {

    }

    override fun onItemLongClick(position: Int, view: View?): Boolean {
        return false
    }

    abstract override fun getLayoutId(): Int

    abstract fun getAdapter(): A
}